core/scripts/cli/sponsor/create.js

```diff
@@ -27,6 +27,16 @@ const create = async (web3, artifacts, emp) => {
   const collateralNeeded = toBN(tokens)
     .mul(gcr)
     .divRound(scalingFactor);
+  // TODO: In some cases, collateralNeeded/tokens can be < gcr, which causes the `create` to fail.
+  // Why!?
```

Comment:
@nicholaspai The create fails in some cases because the amount of collateral the scripts thinks we need isn't actually enough (less by 1 wei). I *thought* we were handling the rounding behavior correctly/conservatively here. Any suggestions on what we should do instead?

Reply:
do `mul` and `divRound` both floor their outputs?

Reply:
I don't think mul causes any rounding, and presumably divRound rounds up (while the contract always truncates). But clearly something is messed up.

If you want to reproduce, use the CLI and keep entering in weird amounts for create/redeem, like 0.234123123 tokens.

Reply:
I'm trying to reproduce, what was the GCR when you ran it?

Reply:
are you using the convenience scripts I created?

Reply:
I did use DeployEMP.js to set it up.

A sample where it got stuck (I tried to create 0.235 tokens):

```
totalPositionCollateral= 15892977142857142857
totalTokens= 8000000000000000000
gcr= 1986622142857142857
collateralNeeded= 466856203571428571
You'll need 0.466856203571428571 ETH to borrow tokens
``` 

You actually end up needing 1 wei more ETH (computations done in Python):

```
actual GCR = 1986622142857142857
collateralNeeded / 0.235 = 1986622142857142855
```

Which is a little bit less.

The actual amount of collateral you'd actually need is (GCR * numTokens):

```
> 1986622142857142857 * (235 * 10**15)
466856203571428571395000000000000000
```
The last part of it is 571.395.

The only option I've been able to come up with is to use a better big number library than what comes with web3 that supports divCeil or an equivalent (https://github.com/MikeMcl/bignumber.js/). Any other ideas you can think of? We used to use BigNumber.js in the old sponsor dapp for something, so that's definitely still a valid option.

Reply:
I just repeated your calculations in truffle console and compared with the "expected" calculations using wolfram alpha, and agree that the `divRound` chops off the last 3 LSD's: 395. 

Do you think this is because `bn.js` has limited precision? Or is `divRound` just always flooring this result? I think its worth exploring bignumber.js

Reply:
Fixed with BigNumber.js.